src/qt/graphics.cpp

```diff
@@ -169,29 +140,123 @@ class WXDLLIMPEXP_CORE wxQtPenData : public wxGraphicsObjectRefData
+            {
+                wxAlphaPixelData::Iterator rowStart = p;
+                wxUint32* const rowStartDst = data;
+                for ( int x = 0; x < pixData.GetWidth(); ++x )
+                {
+                    unsigned char alpha = hasAlpha ? p.Alpha() : wxALPHA_OPAQUE;
+                    // Bitmap pixel bits are already premultiplied.
+                    *data = (alpha << 24 | p.Red() << 16 | p.Green() << 8 | p.Blue());
+                    ++data;
+                    ++p;
+                }
+
+                data = rowStartDst + stride / 4;
+                p = rowStart;
+                p.OffsetY(pixData, 1);
+            }
+        }
+        else // no alpha
+        {
+            wxNativePixelData pixData(bmpSource);
+            wxCHECK_RET( pixData, "Failed to gain raw access to bitmap data." );
+
+            wxNativePixelData::Iterator p(pixData);
+            for ( int y = 0; y < pixData.GetHeight(); ++y )
+            {
+                wxNativePixelData::Iterator rowStart = p;
+                wxUint32* const rowStartDst = data;
+                for ( int x = 0; x < pixData.GetWidth(); ++x )
+                {
+                    *data = (wxALPHA_OPAQUE << 24 | p.Red() << 16 | p.Green() << 8 | p.Blue() );
+                    ++data;
+                    ++p;
+                }
+
+                data = rowStartDst + stride / 4;
+                p = rowStart;
+                p.OffsetY(pixData, 1);
+            }
+        }
+
+        // if there is a mask, set the alpha bytes in the target buffer to
+        // fully transparent or retain original value
+        if ( bmp.GetMask() )
+        {
+            wxBitmap bmpMask = bmp.GetMask()->GetBitmap();
+            data = reinterpret_cast<wxUint32*>(qtImage.bits());
+            wxNativePixelData pixData(bmpMask);
+            wxCHECK_RET( pixData, "Failed to gain raw access to mask data." );
+
+            wxNativePixelData::Iterator p(pixData);
+            for ( int y = 0; y < pixData.GetHeight(); ++y )
+            {
+                wxNativePixelData::Iterator rowStart = p;
+                wxUint32* const rowStartDst = data;
+                for ( int x = 0; x < pixData.GetWidth(); ++x )
+                {
+                    if ( p.Red()+p.Green()+p.Blue() == 0 )
+                        *data = 0;
+
+                    ++data;
+                    ++p;
+                }
+
+                data = rowStartDst + stride / 4;
+                p = rowStart;
+                p.OffsetY(pixData, 1);
+            }
+        }
+
+        m_pixmap = QPixmap::fromImage(qtImage);
     }
 
-    QPixmap* GetPixmap() const
+    QPixmap GetPixmap() const
     {
-        return m_pixmap;
+        return const_cast<wxQtBitmapData*>(this)->m_pixmap;
     }
 
     virtual void* GetNativeBitmap() const override
     {
-        return m_pixmap;
+        auto This = const_cast<wxQtBitmapData*>(this);
+        return &This->m_pixmap;
```

Comment:
Why not

```suggestion
        return const_cast<QPixmap*>(&m_pixmap);
```

which would be more clear IMO?

If not, I'd like to at least rename `This` to `self` for consistency with the same casts done elsewhere, but this I can do myself.